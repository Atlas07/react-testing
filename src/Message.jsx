import React from 'react';
import { string } from 'prop-types';

const Message = ({ type, data }) => (
  <p>{data}</p>
);

Message.propTypes = {
  type: string.isRequired,
  data: string.isRequired,
};

export default Message;
