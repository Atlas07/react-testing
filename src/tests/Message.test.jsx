import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Message from '../Message';

it('Renders without crashing', () => {
  shallow(<Message type="info" data="Loading" />);
});

it('render correctly message component', () => {
  const MessageComponent = renderer.create(<Message data="Loading..." type="info" />).toJSON();
  expect(MessageComponent).toMatchSnapshot();
});

it('Have required props', () => {
  const props = {
    data: 'Loading...',
    type: 'info',
  };

  const wrapper = mount(<Message {...props} />);

  expect(wrapper.prop('data')).toEqual('Loading...');
  expect(wrapper.prop('type')).toEqual('info');
});