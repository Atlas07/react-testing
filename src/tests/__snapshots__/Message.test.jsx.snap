// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly message component 1`] = `
<p>
  Loading...
</p>
`;
